### PR TITLE
Add more node versions for CI

### PR DESCRIPTION
**Pull request by JakeChampion**
This adds the LTS versions of node and the latest stable versions into Travis CI for testing against

### CODE REVIEW

**Comment by ljharb** [comment]
Looks like some of the tests fail. Can we include the fixes?

**Comment by JakeChampion** [comment]
Some of these engines will work if we separate out the feature detection from the polyfill. I'll look into what will make the other engines work.

**Comment by JakeChampion** [comment]
Actually, I think all of them will work if we separate the feature detect from the polyfill.

**Comment by JakeChampion** [comment]
I've merged the no-guard branch into this branch to check if that wold make the engines all pass the tests. It seems node v5.0 fails on installing npm -- `npm WARN checkPermissions Missing write access to /home/travis/.nvm/versions/node/v5.0.0/lib/node_modules/npm/sha/node_modules` -- I'm not sure what to do about that.

**Comment by JakeChampion** [comment]
It now passes 18 of the 20 engines, the 2 it doesn't pass have been failing for some time and are in the "allowed failures" section.

File: .travis.yml [hunk 1]
```diff
@@ -16,19 +21,10 @@ node_js:
   - "0.6"
   - "0.4"
 before_install:
-  - '[ "${TRAVIS_NODE_VERSION}" == "0.6" ] || npm install -g npm@~1.4.6'
+  - '[ "${TRAVIS_NODE_VERSION}" == "0.6" ] || [ "${TRAVIS_NODE_VERSION}" == "v5.0" ] || npm install -g npm@~1.4.6'
```
Comment (ljharb) [inline .travis.yml]: I've found [this combination](https://github.com/ljharb/is-callable/blob/master/.travis.yml#L34-L36) to be most effective.

**Comment by ljharb** [comment]
Pending a rebase, and tweaking the before_install, I think this will be good to go as well!

**Comment by JakeChampion** [comment]
@ljharb Updated :-)

**Comment by ljharb** [comment]
Thanks!

**Comment by JakeChampion** [comment]
@ljharb Would it be possible for these changes to be published as a release to npmjs' registry?

**Comment by ljharb** [comment]
Sure, I'll take a look at it soon

**Comment by ljharb** [comment]
@mathiasbynens looks like you'll need to do the publishing since i'm not an owner on npm :-)

**Comment by mathiasbynens** [comment]
@ljharb Added you as owner! (Should’ve done that a long time ago; apologies.)

**Comment by ljharb** [comment]
Thanks! @JakeChampion, I'll publish these updates today.

**Comment by ljharb** [comment]
Published as `v0.3.0`, with a changelog moving forward :-)

**Pull request merged**